utils/text_formatting.tsx

```diff
@@ -239,9 +239,16 @@ export function formatText(
              * the Markdown.format function removes all duplicate line-breaks beforehand, so it is safe to just
              * replace occurrences which are not followed by opening <p> tags to prevent duplicate line-breaks
              *
-             * @link to regex101.com: https://regex101.com/r/iPZ02c/1
+             * The data-codeblock-code part is a fix for MM-45349 - do not replace newlines with `<br/>`
+             * in code blocks, as they become visible in the message
              */
-            output = output.replace(/[\r\n]+(?!(<p>))/g, '<br/>');
+            output = output.replace(/data-codeblock-code="[^"]+"|[\r\n]+(?!(<p>))/g, (match: string) => {
```

Comment:
To make sure I understand this correctly, it captures `data-codeblock-code="..."` first meaning that it won't replace any newlines inside that piece of text? If so, that's pretty interesting.

Reply:
Yes, exactly. Because `data-codeblock-code` is converted into a `<code>` block, the newlines (`<br/>`) became visible. `\n` inside the `<code>` blocks are transformed into new lines in the UI, so no need to touch them.